### PR TITLE
Move debug functionality to a --debug flag

### LINKED ISSUE #1082: Browser/VM debug modes should be enableable outside of --pause-after-load
As of right now, debug modes are controlled primarily by the `--pause-after-load` flag, and this will make it difficult in the future to control debug-related functionality, such as gathering code coverage. For this reason, I believe it would be best to decouple these modes from `--pause-after-load` in favor of a more adaptable API.

This could be done either _just_ through the API, or could even be added to its own flag, such as `--debug`. I don't see much reason to _not_ add it to its own flag, so that might be the better solution.

Here's where those debug modes are set:
 - [In vm/platform.dart](https://github.com/dart-lang/test/blob/77e09b4e9aa5e1a3e14191e368b1c7f5536808a5/pkgs/test_core/lib/src/runner/vm/platform.dart#L64)
- [In browser/platform.dart](https://github.com/dart-lang/test/blob/48a0d09e1b77e5e80e6b18eca6cf32626c6b59b9/pkgs/test/lib/src/runner/browser/platform.dart#L400)
- [Again in browser/platform.dart](https://github.com/dart-lang/test/blob/48a0d09e1b77e5e80e6b18eca6cf32626c6b59b9/pkgs/test/lib/src/runner/browser/platform.dart#L405)
- [In reporter/json.dart](https://github.com/dart-lang/test/blob/48a0d09e1b77e5e80e6b18eca6cf32626c6b59b9/pkgs/test_core/lib/src/runner/reporter/json.dart#L179)
- [And in an important alert in runner.dart](https://github.com/dart-lang/test/blob/1ccf56082adf35d5436e09793f547dbaa2e48218/pkgs/test_core/lib/src/runner.dart#L101)

### PR DESCRIPTION
**Pull request by willdrach-wk**
Fixes #1082

### CODE REVIEW

File: pkgs/test/test/runner/runner_test.dart [hunk 1]
```diff
@@ -91,6 +91,8 @@ Usage: pub run test [files or directories...]
     --pause-after-load            Pauses for debugging before any tests execute.
                                   Implies --concurrency=1 and --timeout=none.
                                   Currently only supported for browser tests.
```
Comment (evanweible-wf) [inline pkgs/test/test/runner/runner_test.dart]: Should add to this help text that this flag implies `--debug`

File: pkgs/test_core/lib/src/runner/configuration/args.dart [hunk 2]
```diff
@@ -82,9 +82,12 @@ final ArgParser _parser = (() {
       defaultsTo: '30s');
   parser.addFlag("pause-after-load",
       help: 'Pauses for debugging before any tests execute.\n'
-          'Implies --concurrency=1 and --timeout=none.\n'
+          'Implies --concurrency=1, --debug, and --timeout=none.\n'
           'Currently only supported for browser tests.',
       negatable: false);
+  parser.addFlag("debug",
+      help: 'Runs browsers in debug mode.',
```
Comment (evanweible-wf) [inline pkgs/test_core/lib/src/runner/configuration/args.dart]: Is this accurate for browsers that aren't Chrome?

Comment (willdrach-wk) [reply pkgs/test_core/lib/src/runner/configuration/args.dart]: Updated

File: pkgs/test/test/runner/runner_test.dart (hunk 1 above)
Comment (willdrach-wk) [reply pkgs/test/test/runner/runner_test.dart]: Updated

**Comment by grouma** [comment]
Note I'll hold off on reviewing this until you ping me / remove the WIP tag.

**Comment by willdrach-wk** [comment]
@grouma this should be ready for a first pass of review

**Comment by willdrach-wk** [comment]
Changelogs updated

File: pkgs/test/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 1.6.12
```
Comment (grouma) [inline pkgs/test/CHANGELOG.md]: [nit] We'll want to go with 1.7.0 since this is technically a new feature.

**Comment by willdrach-wk** [comment]
@grouma [CONTRIBUTING.md](https://github.com/dart-lang/test/blob/master/CONTRIBUTING.md) says not to format, is that out of date?

File: pkgs/test/CHANGELOG.md (hunk 3 above)
Comment (willdrach-wk) [reply pkgs/test/CHANGELOG.md]: 👍 Done

**Comment by grouma** [comment]
> @grouma CONTRIBUTING.md says not to format, is that out of date?

Super out of date!

**Comment by willdrach-wk** [comment]
Good to know!

**Comment by jakemac53** [comment]
> @grouma CONTRIBUTING.md says not to format, is that out of date?

:rofl:

**Comment by willdrach-wk** [comment]
This is ready whenever you guys are ready @grouma @jakemac53

**Pull request merged**